src/main/java/de/tum/in/www1/artemis/service/connectors/localci/scaparser/strategy/CheckstyleParser.java

```diff
@@ -1,62 +1,87 @@
 package de.tum.in.www1.artemis.service.connectors.localci.scaparser.strategy;
 
-import org.w3c.dom.Document;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlElementWrapper;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
 
 import de.tum.in.www1.artemis.domain.enumeration.StaticCodeAnalysisTool;
+import de.tum.in.www1.artemis.service.dto.StaticCodeAnalysisIssue;
 import de.tum.in.www1.artemis.service.dto.StaticCodeAnalysisReportDTO;
-import de.tum.in.www1.artemis.service.dto.StaticCodeAnalysisReportDTO.StaticCodeAnalysisIssue;
 
-/**
- * Parser for the Checkstyle static code analysis tool.
- */
-class CheckstyleParser extends CheckstyleFormatParser {
+@JsonIgnoreProperties(ignoreUnknown = true)
+record CheckstyleFile(@JacksonXmlProperty(isAttribute = true, localName = "name") String name,
+
+        @JacksonXmlElementWrapper(useWrapping = false) @JacksonXmlProperty(localName = "error") List<CheckstyleError> errors) {
+}
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+record CheckstyleError(@JacksonXmlProperty(isAttribute = true, localName = "line") Integer line,
+
+        @JacksonXmlProperty(isAttribute = true, localName = "column") Integer column,
+
+        @JacksonXmlProperty(isAttribute = true, localName = "severity") String severity,
+
+        @JacksonXmlProperty(isAttribute = true, localName = "message") String message,
+
+        @JacksonXmlProperty(isAttribute = true, localName = "source") String source) {
+}
```

Comment:
Do we need to use the wrapper `Integer` instead of `int` here?

Reply:
Yes, the column is optional.